components/data-feed-service/service/data-feed-poll-task.go

```diff
@@ -0,0 +1,209 @@
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	cfgmgmtRequest "github.com/chef/automate/api/interservice/cfgmgmt/request"
+	cfgmgmt "github.com/chef/automate/api/interservice/cfgmgmt/service"
+	"github.com/chef/automate/components/data-feed-service/config"
+	"github.com/chef/automate/components/data-feed-service/dao"
+	"github.com/chef/automate/lib/cereal"
+	"github.com/chef/automate/lib/grpc/secureconn"
+)
+
+var (
+	dataFeedPollTaskName = cereal.NewTaskName("data-feed-poll")
+)
+
+type DataFeedPollTask struct {
+	cfgMgmt cfgmgmt.CfgMgmtClient
+	db      *dao.DB
+	manager *cereal.Manager
+}
+
+type DataFeedPollTaskParams struct {
+	AssetPageSize   int32
+	ReportsPageSize int32
+	FeedInterval    time.Duration
+	NextFeedStart   time.Time
+	NextFeedEnd     time.Time
+}
+
+type DataFeedPollTaskResults struct {
+	NodeIDs   map[string]NodeIDs
+	FeedStart time.Time
+	FeedEnd   time.Time
+}
+
+// ClientID the node ID assigned by chef client
+// ComplianceID the node ID in a compliance report
+type NodeIDs struct {
+	ClientID     string
+	ComplianceID string
+}
+
+func NewDataFeedPollTask(dataFeedConfig *config.DataFeedConfig, connFactory *secureconn.Factory, db *dao.DB, manager *cereal.Manager) (*DataFeedPollTask, error) {
+
+	cfgMgmtConn, err := connFactory.Dial("config-mgmt-service", dataFeedConfig.CfgmgmtConfig.Target)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not connect to config-mgmt-service")
+	}
+
+	return &DataFeedPollTask{
+		cfgMgmt: cfgmgmt.NewCfgMgmtClient(cfgMgmtConn),
+		db:      db,
+		manager: manager,
+	}, nil
+}
+
+func (d *DataFeedPollTask) Run(ctx context.Context, task cereal.Task) (interface{}, error) {
+	params := DataFeedWorkflowParams{}
+	err := task.GetParameters(&params)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse task parameters")
+	}
+	log.Debugf("DataFeedPollTask.Run params %v", params)
+	log.Debugf("DataFeedPollTask.Run FeedStart %v", params.FeedStart)
+	log.Debugf("DataFeedPollTask.Run FeedEnd %v", params.FeedEnd)
+	log.Debugf("DataFeedPollTask.Run NextStart %v", params.PollTaskParams.NextFeedStart)
+	log.Debugf("DataFeedPollTask.Run NextEnd %v", params.PollTaskParams.NextFeedEnd)
+	now := time.Now()
+	params = d.getFeedTimes(params, now)
+	feedStartTime := params.PollTaskParams.NextFeedStart
+	feedEndTime := params.PollTaskParams.NextFeedEnd
+	destinations, err := d.db.ListDBDestinations()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get destinations from db")
+	}
+	params.FeedStart = feedStartTime
+	params.FeedEnd = feedEndTime
+	params.PollTaskParams.NextFeedStart = feedEndTime
+	params.PollTaskParams.NextFeedEnd = feedEndTime.Add(params.PollTaskParams.FeedInterval)
+
+	if len(destinations) == 0 {
+		log.Info("DataFeedPollTask.Run no destinations returning")
+		return nil, nil
+	}
+
+	nodeIDs, err := d.GetChangedNodes(ctx, params.PollTaskParams.AssetPageSize, feedStartTime, feedEndTime)
+	log.Debugf("DataFeedPollTask nodes %v", nodeIDs)
```

Comment:
While this is debug, I think we might still want to avoid log messages that potentially output a boatload of data.  Might we consider just logging the count of nodeIDs?

Reply:
I'll look through the debug, I meant to take this out. There are probably a few others like it.